src/format.cc

```diff
@@ -55,7 +55,7 @@ vformat_to(buffer<char>&, string_view,
                type_identity_t<char>>>);
 }  // namespace detail
 
-template struct FMT_INSTANTIATION_DEF_API detail::basic_data<void>;
+template struct detail::basic_data<void>;
```

Comment:
Please add a short comment saying that clang doesn't allow `__declspec(dllexport)` here.